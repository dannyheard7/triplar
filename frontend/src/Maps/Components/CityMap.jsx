import React from 'react'
import {Map, TileLayer} from 'react-leaflet'

import 'Maps/styles/leaflet/leaflet.css';
import 'Maps/styles/maps.css';

export default class CityMap extends React.Component {

  render() {
    const position = [this.props.city.location.lat, this.props.city.location.lng];

    return (
        <div>
          <Map center={position} zoom={13}>
            <TileLayer
              attribution="<a href='https://wikimediafoundation.org/wiki/Maps_Terms_of_Use'>Wikimedia</a>"
              url="https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}{r}.png"
            />
          </Map>
        </div>
    )
  }
}
