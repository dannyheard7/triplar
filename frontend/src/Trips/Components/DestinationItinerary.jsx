import React from "react";
import Moment from 'moment';
import { daysBetweenDates } from 'utils/utils.js';

import CityMap from "Maps/Components/CityMap";


export default class DestinationItinerary extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            showMap: false,
        };

        this.onClick = this.onClick.bind(this);
    }

    onClick() {
        this.setState({showMap: !this.state.showMap});
    }

    renderItineraryDay(day) {
        return (
            <p key={day}>{day}</p>
        );
    }

    render () {
        const itinerary = this.props.itinerary;
        const start_date = Moment(itinerary.start_date);
        const end_date = Moment(itinerary.end_date);

        const days = daysBetweenDates(start_date, end_date).map(this.renderItineraryDay);

        return (
            <div>
                <div className="row">
                    <h3 onClick={this.onClick} style={{cursor: "pointer", color: "blue"}}>{itinerary.city.name_std}</h3>
                    <p>{start_date.format('dddd Do MMMM') } - {end_date.format('dddd Do MMMM') }</p>
                </div>
                {this.state.showMap && <CityMap city={itinerary.city} /> }
                {days}
            </div>
        )
    }
}

DestinationItinerary.defaultProps = {
    itinerary: {}
};
