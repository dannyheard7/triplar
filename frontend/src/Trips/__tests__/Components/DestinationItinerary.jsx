import React from "react";

import DestinationItinerary from "Trips/Components/DestinationItinerary";
import ShallowRenderer from "react-test-renderer/shallow";
import {shallow} from "enzyme/build";
import CityMap from "Maps/Components/CityMap";

const renderer = new ShallowRenderer();


describe('<DestinationItinerary />', () => {
    const props = {
        itinerary: {
            city: {
                name_std: "Bristol",
                country: "United Kingdom"
            },
            start_date: "2018-08-10",
            end_date: "2018-08-12",
        }
    };

    test('<DestinationItinerary /> renders correctly', () => {
        const result = renderer.render(<DestinationItinerary {...props} />);
        expect(result).toMatchSnapshot();
    });

    test('<CityMap /> is shown when showMap is true', () => {
         const container = shallow(<DestinationItinerary  {...props} />);
         container.setState({showMap: true});

         expect(container.find(CityMap).length).toEqual(1);
    });
});