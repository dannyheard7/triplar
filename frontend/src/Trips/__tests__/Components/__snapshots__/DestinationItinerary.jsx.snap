// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DestinationItinerary /> <DestinationItinerary /> renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <h3
      onClick={[Function]}
      style={
        Object {
          "color": "blue",
          "cursor": "pointer",
        }
      }
    >
      Bristol
    </h3>
    <p>
      Friday 10th August
       - 
      Sunday 12th August
    </p>
  </div>
  <p>
    Friday 10th August
  </p>
  <p>
    Saturday 11th August
  </p>
  <p>
    Sunday 12th August
  </p>
</div>
`;
